webapp/channels/src/components/status_icon_new.tsx

```diff
@@ -4,32 +4,25 @@
 import React from 'react';
 
 type Props = {
-    className: string;
-    status: string;
+    className?: string;
+    status?: string;
 }
 
-export default class StatusIconNew extends React.PureComponent<Props> {
-    static defaultProps: Props = {
-        className: '',
-        status: '',
-    };
-
-    render() {
-        const {status, className} = this.props;
+const StatusIconNew = ({className = '', status = ''}: Props) => {
+    if (!status) {
+        return null;
+    }
 
-        if (!status) {
-            return null;
-        }
+    let iconName = 'icon-circle-outline';
+    if (status === 'online') {
+        iconName = 'icon-check-circle';
+    } else if (status === 'away') {
+        iconName = 'icon-clock';
+    } else if (status === 'dnd') {
+        iconName = 'icon-minus-circle';
+    }
```

Comment:
Instead of using if else we can use an object ! like may be in future new status may get added right? so it will become messy ! @larkox what is ur opinion
```ts
const statusToIconMap = {
    'online': 'icon-check-circle',
    'away': 'icon-clock',
    'dnd': 'icon-minus-circle'
    'default': 'icon-circle-outline'
}
```

Reply:
@larkox Yes